### ISSUE TITLE
Make site theme-able so it's easy to do a rebrand and used by others

### ISSUE THREAD

**Issue by epicfaace** [TreeHacks/application-portal-frontend#49]

**Comment by oliviabrown9** [comment]
I've been thinking about this a lot and think it could be a *huge* offering to the hackathon community to make this more usable for others. I think

- theming
- settings reviewers in admin panel
- selecting reviewing criteria
- turning on/off travel tab (as well as maybe setting up bus routes in the admin panel)
- turning on/off room reservation system
- turning off login with Stanford
- more and better documentation

would be amazing. Also, a simple deploy to Heroku button would be incredible and likely help people actually use it. I think the current hackathon registration systems are lacking so much of the value of the portal (hence why we built this one) and we could help a lot of hackathons if we have time over the next few months to prioritize making this amazing.

**Comment by epicfaace** [comment]
@theopolisme 

A few architecture questions we'd need to address:

- Does it still make sense to have two git repos, application-portal and application-portal-frontend? 
  - **Pro:** I split the repos because I envisioned that the API could be deployed separately from the frontend and used for different frontends/websites.
  - **Con:** However, it would be much easier for others to deploy and develop on if it were all in one repo. Additionally, the "frontend" is just a static HTML site.
- We have several external dependencies outside of Heroku, namely, AWS Cognito for authentication, AWS S3 for resume storage, and AWS SES for emails. Does it still make sense to keep these AWS dependencies?
  - **Pro:** AWS is cheap, almost free.
  - **Con:** Requires more setup, no longer simply a "deploy to heroku" option
  - Alternatively, we could accomplish the same functionality using [SendGrid](https://elements.heroku.com/addons/sendgrid) (emails), [Bucketeer](https://elements.heroku.com/addons/bucketeer) (S3), our own custom authentication or [Auth0](https://elements.heroku.com/addons/auth0) rather than Cognito. But these cost more money than just AWS services.

Personally, I would think it best to 1) merge both frontend and backend into one repo, and 2) keep using AWS but just have people go through the extra setup (maybe use AWS Cloudformation to simplify the pain). Let me know what you think, though.

**Comment by theopolisme** [comment]
1. I agree & think best to bundle into a single repository and then (semi-related) have lots of instructions for an easy "drop-in" edge caching layer.

2. Ideal solution imo is decoupling AWS specific code. Possibly modularizing such that you can choose to use AWS or choose to use something else (e.g. making it more pluggable). We're already pretty close to that -- there is not much AWS-specific code mixed in with business logic, it's mostly all abstracted out. Allowing user to set environ variables or perhaps app.json file (through the heroku one-click) and then having system cleverly choose what to use (e.g. is SendGrid API key set? Use that for emails. SES? Ok, no prob, I'll use that) would be holy grail.

**Comment by theopolisme** [comment]
We'll also have a super-easy Eventive integration ;))

**Comment by epicfaace** [comment]
When you mention a "1. "drop-in" edge caching layer", what do you have in mind -- something like AWS CloudFront?

**Comment by theopolisme** [comment]
Right, but ideally pluggable enough that we could have instructions for setting up something non-AWS like CloudFlare too :)

**Comment by epicfaace** [comment]
I think the best way to handle this kind of configuration would be rather than creating a UI for config ourselves, to store config in a JSON/YAML file. Imagine that these would be the steps:
- Run `npm install create-root-app`
- This installs `root` as a node dependency, then creates a `config.yaml` file in the root directory, which one can then change theming, application questions, etc.
- One could potentially also add particular HTML/JSX files to override templates.

The YAML file could be structured something like this:

```
colors:
    primary: #fff
    secondary: #eee
questions:
    age: ...
    q1: ....
deadlines:
    early: 5/5/2019
```
...and so on
